File: packages/shared-types/src/set.ts
/**
 * Set-related TypeScript types matching API schemas (snake_case).
 * These types match the JSON responses from the FastAPI backend.
 */

export interface ApiSet {
  // Identifiers
  id: string;
  name: string;
  series: string;

  // Release info
  release_date?: string | null;
  release_date_jp?: string | null;
  card_count?: number | null;

  // Images
  logo_url?: string | null;
  symbol_url?: string | null;

  // Legalities
  legalities?: Record<string, string> | null;

  // Timestamps
  created_at: string;
  updated_at: string;
}

File: apps/api/src/schemas/set.py
"""Set-related Pydantic schemas."""

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class SetResponse(BaseModel):
    """Full schema for card set data."""

    model_config = ConfigDict(from_attributes=True)

    # Identifiers
    id: str
    name: str
    series: str

    # Release info
    release_date: date | None = None
    release_date_jp: date | None = None
    card_count: int | None = None

    # Images
    logo_url: str | None = None
    symbol_url: str | None = None

    # Legalities
    legalities: dict[str, str] | None = None

    # Timestamps
    created_at: datetime
    updated_at: datetime
